src/FffGcodeWriter.cpp

```diff
@@ -2379,159 +2380,187 @@ bool FffGcodeWriter::addSupportToGCode(const SliceDataStorage& storage, LayerPla
+        Polygons support_polygons;
+        Polygons support_lines;
+        for (unsigned int density_idx = part.infill_area_per_combine_per_density.size() - 1; (int)density_idx >= 0; density_idx--)
+        {
+            if (combine_idx >= part.infill_area_per_combine_per_density[density_idx].size())
+            {
+                continue;
+            }
 
-                constexpr coord_t pocket_size = 0;
+            const Polygons& area = part.infill_area_per_combine_per_density[density_idx][combine_idx];
+            const coord_t area_offset = (config.first_layer && config.pattern == EFillMethod::CONCENTRIC) ? (line_width * config.brim_line_count / 1000) : 0; // Todo check if brim_line  / 1000 is correct
+            const Polygons area_with_offset = area_offset > 0 ? area.offset(area_offset) : Polygons();
 
-                Infill infill_comp(support_pattern, zig_zaggify_infill, connect_polygons, support_area_offset > 0 ? support_area_with_offset : support_area, offset_from_outline, support_line_width,
-                                   support_line_distance_here, current_support_infill_overlap, infill_multiplier, support_infill_angle, gcode_layer.z, support_shift, wall_line_count, infill_origin,
-                                   perimeter_gaps, infill_extruder.settings.get<bool>("support_connect_zigzags"), use_endpieces,
-                                   skip_some_zags, zag_skip_count, pocket_size);
-                infill_comp.generate(support_polygons, support_lines, storage.support.cross_fill_provider);
-                if (support_area_offset > 0)
-                {
-                    // Only polygons (not lines), because this only happens in the case of concentric on layer 0 (if support brim is enabled).
-                    support_polygons = support_polygons.difference(support_area_with_offset.difference(support_area.offset(support_line_width / 2)));
-                }
+            const unsigned int density_factor = 2 << density_idx; // == pow(2, density_idx + 1)
+            coord_t line_distance_here = config.line_distance * density_factor; // the highest density infill combines with the next to create a grid with density_factor 1
+            const coord_t support_shift = line_distance_here / 2;
+            if (density_idx == part.infill_area_per_combine_per_density.size() - 1 || config.pattern == EFillMethod::CROSS || config.pattern == EFillMethod::CROSS_3D)
+            {
+                line_distance_here /= 2;
             }
 
-            if (support_lines.size() > 0 || support_polygons.size() > 0)
+            constexpr coord_t offset_from_outline = 0;
+            constexpr bool use_endpieces = true;
+            constexpr Polygons* perimeter_gaps = nullptr;
+            const Point infill_origin;
+            const Polygons in_outline = area_offset > 0 ? area_with_offset : area;
+
+            constexpr coord_t pocket_size = 0;
+            constexpr coord_t wall_line_count = 0;
```

Comment:
Previously we were using the `Infill` class with the appropriate `wall_line_count` to implement the correct number of walls around the support. The point of this was that most places where the infill patterns get used can also have extra walls around it: support infill, skin, and normal infill. If there needs to be a change, you'd then only need to implement that change in one spot.

You've set `wall_line_count` to 0 however. Why did you implement the extra wall here in `FffGcodeWriter` and not in the `Infill` class, where the implementation could get re-used for the skin and infill? I think this is going to introduce needless code duplication and more work to implement the other "extra wall" settings, and also add on to the `FffGcodeWriter` more logic which is not related to the orchestrating role of the class (which is already a problem for many other things in the class, I know). It'll also make the `wall_line_count` parameter unused code.